src/shared/bootspec.c

```diff
@@ -1392,13 +1392,141 @@ static void boot_entry_file_list(
                 *ret_status = status;
 }
 
+static int copy_cmdline(
+                char *cmdline,
+                char **ret) {
+
+        assert(ret);
+
+        _cleanup_free_ char *temp = *ret;
+
+        *ret = strnappend(temp, cmdline, strlen(cmdline));
+        if (temp)
+                free(temp);
```

Comment:
free() is happy to accept NULL pointers, and is a nop then. hence no need for the conditionalization.